### DISCUSSION TITLE
Stuck on Checking Availability

### DISCUSSION THREAD

**Discussion by dan296** [community/community#53908] [Repositories]
Is anyone else currently having this issue? I am attempting to create a github repo through the "Create a new repository" form on the web application, but it seems the form gets stuck attempting to check the availability of the repository name. I inspected the page in Chrome and found a failing POST call to github.com/repositories/check-name. I believe this bug must have just come about

**Comment by GabrielBoetto** [comment]
I have the same problem..

**Comment by multikitty** [comment]
In some cases, checking the availability of the repository name is delayed because of network connection issues.
Please check your network and refresh page and try again.
Hope this will help you.

**Comment by thispaul** [comment]
Hi, this issue should now be resolved. Please let us know if you are still having trouble creating a repo.

**Reply by tarazansafak** [reply]
seems like its still the same

**Reply by nikorter** [reply]
same here

**Reply by jamiejoevanstone** [reply]
Looks like there is still an issue with this. I'm getting a 500 response from this endpoint.`https://github.com/repositories/check-name`.

**Reply by digi-food** [reply]
I've been having this problem for three days now. I've done all the suggested fixes, I'm on a project deadline and this is incredibly frustrating. Are there other suggestions of what to try?

**Reply by Chaoschaot232** [reply]
Error still persisten within Chrome with and without any possible AddOn(s).

Here is a Log from what is shown to me within F12, maybe this helps out to fix it:
[github.com-1709750085383.log](https://github.com/community/community/files/14514163/github.com-1709750085383.log)

And no, I am not going to change my default browser not even Edge ;)

**Comment by Abates-95** [comment]
It was working fine about an hour ago but once again stuck checking availablity name

**Comment by LordXXander** [comment]
Still stuck too

**Comment by zelon** [comment]
Same here

**Comment by yashoswalyo** [comment]
Just check https://www.githubstatus.com/ to know about GitHub site status

**Comment by ShakirKhan777** [comment]
Can someone help pls have been stuck from yesterday with the same issue "Checking availability..." ?

**Comment by NcFerrari** [comment]
Use another browser (i use opera, but for creating new repo i used chrome and it worked)

**Comment by Flashxyz** [comment]
clear your history and all cookies that always works

**Comment by dhivyapriya1998** [comment]
still the issue

**Comment by tuanluu-agilityio** [comment]
Got same issue right now
![image](https://github.com/community/community/assets/24542782/e3a60af4-723a-46c7-be10-a404464c1090)

**Reply by amiyading** [reply]
Change other browser

**Reply by tuanluu-agilityio** [reply]
Change Chrome -> Edge browser working fine for me. Many Thanks @amiyading

**Comment by mr-bronson** [comment]
This is not a matter of "use a different browser", which is a dumb answer anyway, if you're already using a mainstream browser such as Firefox, Chromium-based (including Edge) or Safari.

This (at least in my case) is due to a github.com bug, in which the developers made the bad assumption that users enable the Origin header.  (The Origin header is one of the dumbest most obvious privacy holes ever engineered, letting those who host common scripts and fonts track your IP address all over the web.)

The specific request, which isn't even cross origin, thus missing the point of CORS, is a POST request to:
`https://github.com/repositories/check-name`
with a form payload containing the proposed repository name, the username, and json_response=true.

What the server returns is a [422 Unprocessable Content](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) error, not in JSON but in HTML, which of course you never see, that says

> Your browser did something unexpected. Please try again. If the error continues, try disabling all browser extensions.

Not a helpful answer, even if you could see it.

It also seems like a poor UI decision to keep you from submitting the page until you get back this very temporary confirmation that the name is available.  It could very well become unavailable between when you check and when you submit the form with the "Create fork" button, so it needs to tell you after it's already tried and failed, not just stand in the way until you pass the little pre-flight check.

A couple workarounds, until GitHub fixes this:

Option 1: Create another browser profile just for using GitHub, and in that profile, enable the stupid Origin header, e.g. in Firefox go to about:config and change "network.http.sendOriginHeader" to 1 or 2 instead of 0.

Option 2: Make a simple browser extension that adds the header in this case.  Code is as follows (tested in Firefox):
[manifest.json](https://github.com/user-attachments/files/16360495/manifest.json)
[background.js.txt](https://github.com/user-attachments/files/16360524/background.js.txt)
(remove the .txt extension - GitHub blocks attaching .js files)

**Comment by Mohsen-graphyy** [comment]
I have the same issue, I have changed the browser and the issue fixed.

**Comment by authifyAlly** [comment]
I agree it's a CORS issue. I started getting the issue when `--disable-web-security --disable-gpu --user-data-dir=~/chromeTemp` was enabled to chrome Target Location. 
Thought it's not a browser issue per-se, the easiest solution is having a separate browser for testing and a separate for rest.

**Comment by mr-bronson** [comment]
For those with Origin header disabled:  If anyone has tried the code I gave in July, you know it's incomplete.  Here's the code I'm currently using.  This allows you to not only pass the name check, but to also continue with the fork.  It also adds the Origin header for the `fetch_and_merge` action that likewise requires it.

[manifest.json](https://github.com/user-attachments/files/17163416/manifest.json)
[background.js.txt](https://github.com/user-attachments/files/17163415/background.js.txt)
(remove the .txt extension - GitHub blocks attaching .js files)

I still don't know what these developers think they're achieving.  My own blasted web host has now started requiring an Origin header to even log in -- some kind of stupid CloudFlare nonsense to check whether or not I'm human, as if bots don't know how to send an Origin header.

**Comment by authifyAlly** [comment]
I think this is solved now. I can see it working as expected.

**Reply by chezbro** [reply]
Still broken for me on Chrome v129.0.6668.71
